### PR TITLE
[WIP] Implement missing API functions

### PR DESCRIPTION
**Pull request by philipflohr**
As written in #7 my work is not yet finished and is not ready to be pulled.

Besides the missing functionality there are no tests. 
Also I'm actually not sure if the remote_query_id flag should default to True or False - I chose False for a simple reason: without explicitly allowing a query with this default value it's only possible to update an attendance record which already exists on the Personio servers.

### CODE REVIEW

**Comment by klamann** [comment]
@philipflohr thanks for your work on this PR. I'm gonna make a short review of the code now and add a few comments.

One thing that would be really important though are tests. The most useful and thorough tests we have right now are in [test_mock_api.py](https://github.com/at-gmbh/personio-py/blob/master/tests/test_mock_api.py), where the response from the server for certain inputs is mocked. Would you be willing to add tests for the API methods you have implemented?

File: src/personio_py/client.py [hunk 1]
```diff
@@ -127,8 +127,7 @@ def request_json(self, path: str, method='GET', params: Dict[str, Any] = None,
                during this request (default: True for json requests)
         :return: the parsed json response, when the request was successful, or a PersonioApiError
         """
-        headers = {'accept': 'application/json'}
-        response = self.request(path, method, params, data, headers, auth_rotation=auth_rotation)
+        response = self.request(path, method, params, data, auth_rotation=auth_rotation)
```
Comment (klamann) [inline src/personio_py/client.py:130]: was there a reason to remove the headers from the request?

File: src/personio_py/models.py [hunk 2]
```diff
@@ -289,6 +289,9 @@ def _check_client(self, client: 'Personio' = None) -> 'Personio':
             client.authenticate()
         return client
 
+    def set_client(self, client: 'Personio'):
+        self._client = client
+
```
Comment (klamann) [inline src/personio_py/models.py:294]: I'm not sure if there is a need for a setter here, this looks like java code to me ^^

We can change `create_attendances()` so that this is not needed

File: src/personio_py/models.py [hunk 3]
```diff
@@ -652,13 +668,49 @@ def to_dict(self, nested=False) -> Dict[str, Any]:
         return d
 
     def _create(self, client: 'Personio'):
-        pass
+        self._client.create_attendances([self])
```
Comment (klamann) [inline src/personio_py/models.py:671]: I left this part unfinished... `client` should have been an optional parameter and the function should choose between `client` and `self._client` or raise an error if both are None. I should probably fix this myself ^^

File: src/personio_py/client.py (hunk 1 above)
Comment (philipflohr) [reply src/personio_py/client.py:130]: setting json= .. instead of data=.. in the request automatically sets the accept header to 'application/json'

File: src/personio_py/client.py [hunk 4]
```diff
@@ -291,28 +289,94 @@ def get_attendances(self, employees: Union[int, List[int], Employee, List[Employ
...
             'company/attendances', Attendance, employees, start_date, end_date)
 
-    def create_attendances(self, attendances: List[Attendance]):
+    def create_attendances(self, attendances: List[Attendance]) -> bool:
         """
-        placeholder; not ready to be used
+        Create all given attendance records.
+
+        Note: Attendances are created sequentially. This function stops on first error.
+        All attendance records before the error will be created, all records after the error will be skipped.
```
Comment (klamann) [inline src/personio_py/client.py:297]: were you able to verify that this works? The [documentation](https://developer.personio.de/reference#post_company-attendances) does not indicate what happens when there are errors in only some of the attendance records.

Comment (philipflohr) [reply src/personio_py/client.py:297]: This is not documented but I verified this using a personio test account

Comment (klamann) [reply src/personio_py/client.py:297]: in this case, maybe we should tell the user which attendances have been created and which have failed? We could return a dictionary like

```
{
  'success': [list of attendance objects that were created],
  'failure': [list of attendance objects that were not created],
}
```

Comment (philipflohr) [reply src/personio_py/client.py:297]: I only did a very quick test of this behavior, but IIRC the response does not contain the information where the error happened.

File: src/personio_py/models.py (hunk 2 above)
Comment (philipflohr) [reply src/personio_py/models.py:294]: Solution proposal: I could change the objects client to be public accessible

**Comment by philipflohr** [comment]
@klamann are the test provided for absences ok like this?

**Comment by philipflohr** [comment]
@klamann can you give any update on this or do you have an idea when you'll have a little time for it?

**Comment by klamann** [comment]
Hey @philipflohr, sorry for letting you wait. I plan to review your changes on the weekend, maybe earlier if I get the chance.
Are you finished with your additions or do you want to add more to this PR?

**Comment by philipflohr** [comment]
There are still test missing for attendances. Could have a quick look into the absences tests? If they are ok I'll provide similar tests for attendances

**Comment by klamann** [comment]
I had a look at the tests; what I like is that there are many tests that work with a live Personio instance and where many edge cases are covered. What I'm missing however are tests that can run as part of the CI pipeline without a live Personio instance, like those defined in [`test_mock_api.py`](https://github.com/at-gmbh/personio-py/blob/master/tests/test_mock_api.py). I know those are more difficult to define, but they are really important to ensure that the API functions are still working when other parts of the implementation are changed.

**Comment by philipflohr** [comment]
I'm closing this PR in favor of multiple smaller PRs.
